src/or/hs_descriptor.c

```diff
@@ -1179,6 +1179,36 @@ desc_encode_v3(const hs_descriptor_t *desc,
 
 /* === DECODING === */
 
+/* Given the token tok for an auth client, decode it as
+ * hs_desc_authorized_client_t. Return 0 on success else -1 on failure. */
```

Comment:
We should document here that the tok->args MUST contain at least 3 elements. The following function doesn't assess that so lets be extra careful.